### DISCUSSION TITLE
How can I stop a desktop application from reopening when clicked on the taskbar icon?

### DISCUSSION THREAD

**Discussion by Pratham95VP** [AvaloniaUI/Avalonia#16074] [Q&A] [answered]
When the application is minimized, I'm looking for a way to prevent it from reopening when clicked on the taskbar icon. From what I understand, this behavior is controlled at the OS level, and while Avalonia may not directly influence it, I'm interested to know if there are any potential workarounds available.

**Comment by workgroupengineering** [comment]
The simplest thing to do is to check if there are any other programs running with the same name and immediately exit the current one.

```c#
    internal class Program
    {
        // Initialization code. Don't use any Avalonia, third-party APIs or any
        // SynchronizationContext-reliant code before AppMain is called: things aren't initialized
        // yet and stuff might break.
        [STAThread]
        public static void Main(string[] args)
        {
            var currentProcess = System.Diagnostics.Process.GetCurrentProcess(); ;
            var currentId = currentProcess.Id;
            var currentProcessName = currentProcess.ProcessName;
            if (System.Diagnostics.Process.GetProcesses().FirstOrDefault(p => p.Id != currentId && p.ProcessName == currentProcessName) is not null)
            {
                return;
            }
            BuildAvaloniaApp()
            .StartWithClassicDesktopLifetime(args);
        }
    }
``` 

There are other methods like mutext but they are not cross platform.

**Reply by Pratham95VP** [reply]
This prevent application from opening new instance when The application is already running . My question is bit different. The application is already running and is in minimized state like shown in the image. A common behavior is when user click on app icon (from taskbar) the app gets reopen or restore its window state. I want to prevent that. 

![Untitled](https://github.com/AvaloniaUI/Avalonia/assets/154887456/4e8a19ca-2086-4880-9d30-6daa87aead67)

**Reply by workgroupengineering** [reply, marked as answer]
Set `ShowInTaskbar="False"` in your Window

**Reply by Pratham95VP** [reply]
> Set `ShowInTaskbar="False"` in your Window

Along with this i have used  https://docs.avaloniaui.net/docs/reference/controls/detailed-reference/tray-icon and provided option to close application.  Thanks..!
